### PR TITLE
Fix losses output shapes

### PR DESCRIPTION
**Pull request by zhanyinx**
## Proposed changes

tf.keras.losses.binary_crossentropy and categorical_crossentropy reduce last dimension of inputs by taking the average over the last dimension. This creates sometimes problem with manipulation of the dimensions (sum, mean over different axis). Here, we solved this issue by artificially expanding the inputs so that tf.keras.losses reduce over this newly created last axis.

 
## Types of changes

 - [x] Bugfix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation content update
 - [ ] Other (please describe):

## Checklist

Put an x in the boxes that apply. You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code so nothing breaks.

 - [x] I have read the [contributing guidelines](https://deepblink.readthedocs.io/en/latest/contributing.html)
 - [x] Lint and unit tests pass locally with my changes (`tox`)
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] I have added necessary documentation (if appropriate)
 - [ ] Any dependent changes have been merged and published in downstream modules
 - [ ] My commit messages format follow the project [git commit message format](https://github.com/slashsbin/styleguide-git-commit-message#commit-message-format)
 - [ ] First time only - I have added myself / the copyright holder to the authors.rst file

💔 Thank you!

### CODE REVIEW

File: deepblink/losses.py [hunk 1]
```diff
@@ -118,20 +137,18 @@ def rmse(y_true, y_pred):
...
-    Optimal value for rmse is 0.
-    Therefore, optimal value for combined_bce_rmse is 0.
-
-    rmse is rescaled with 1/10 to weigh more bce in the calculation of the loss.
+    The optimal values for binary crossentropy (bce) and rmse are both 0.
+    Bce is considered more important so we weighted rmse with 1/10.
     """
-    return binary_crossentropy(y_true[..., 0], y_pred[..., 0]) + rmse(y_true, y_pred) / 10
+    return (
+        binary_crossentropy(y_true[..., 0], y_pred[..., 0]) + rmse(y_true, y_pred) / 10
```
Comment (BBQuercus) [inline deepblink/losses.py:153]: It would probably make sense to not have the indexing `[:, 1:]` in `rmse` but rather pass the proper inputs from this function here.

**Review by BBQuercus — APPROVED** [review]
Updated some docstrings but this looks reasonable. Is there any difference in this way rather than using `K.flatten` and then just allow Keras to do its thing?
